packages/devtools/lib/src/timeline/frames_bar_chart.dart

```diff
@@ -59,6 +81,124 @@ class FramesBarChart extends CoreElement {
       _selectedFrameController.add(selectedFrame?.frame);
     }
   }
+*/
+}
+
+class PlotlyDivGraph extends CoreElement {
+  bool _processDatum;
+  Timer timer;
+  final FramesBarChart framesBarChart;
+  final TimelineFrame frame;
+  final Map<int, TimelineFrame> _frames = {};
+
+  ReadWriteMutex mutex = ReadWriteMutex();
+
+  List<int> dataIndexes = [];
+  List<num> cpuDurations = [];
+  List<num> gpuDurations = [];
+
+  FramesBarPlotly plotlyChart;
+
+  PlotlyDivGraph(this.framesBarChart, this.frame, [bool datum = true])
+      : this._processDatum = datum,
+        super('div') {
+    element.id = 'graph';
+    element.style
+      ..height = '100%'
+      ..width = '100%';
+  }
+
+  int frameIndex = 0;
+  bool _createdPlot = false;
+
+  // This routine should only be called using the mutex.acquireWrite() as it's
+  // destructive to our lists collecting index, and durations which are filled
+  // when onFrameAdded event and we chunk these to the our plotly chart to not
+  // cause the charting to lag.
+  void plotData(TimelineController timelineController) {
+    int dataLength = dataIndexes.length;
+    if (dataLength > 0) {
+      plotlyChart.plotFPSDataList(dataIndexes, cpuDurations,
+          gpuDurations, timelineController.paused);
+
+      dataIndexes.removeRange(0, dataLength);
+      cpuDurations.removeRange(0, dataLength);
+      gpuDurations.removeRange(0, dataLength);
+    }
+  }
+
+  void _plotlyClick(DataEvent data) {
+    int x = data.points[0].x;
+    if (_frames.containsKey(x)) {
+      TimelineFrame timelineFrame = _frames[x];
+      framesBarChart.setSelected(timelineFrame);
+    }
+  }
+
+  void process(
+      TimelineController timelineController, TimelineFrame frame) async {
+    if (!_createdPlot) {
+      plotlyChart = new FramesBarPlotly('graph');
+      plotlyChart.plotFPS();
+
+      _createdPlot = true;
+
+      // Hookup clicks in the plotly chart.
+      plotlyChart.chartClick('graph', _plotlyClick);
+
+      if (!_processDatum) {
+        // Chunk the plots either every 10 frames or every 1 seconds.
+        timer = Timer.periodic(Duration(milliseconds: 1000), (Timer t) async {
```

Comment:
nit:
seconds: 1
instead of
milliseconds: 1000

Would it be possible to increase this update rate? If every 10 frames is ok at 60fps, then we should be able to update plotly  at 6fps.
We might want to tweak the logic to every 20 frames instead of 10 so we don't get overwhelmed if someone uses an ios device that runs at 120fps.

Reply:
Sure I'll change it.